go/vt/tabletserver/query_executor.go

```diff
@@ -106,8 +106,46 @@ func (qre *QueryExecutor) Execute() (reply *mproto.QueryResult) {
 			conn := qre.getConn(qre.qe.connPool)
 			defer conn.Recycle()
 			reply = qre.execSQL(conn, qre.query, true)
+		}
+		if reply != nil {
```

Comment:
Alternate suggestion:
switch {
...
default:
  goto dmlPlan
}
return reply
dmlPlan:
 ...

I think checking for reply != nil can introduce bugs in the future when we switch from panics to error returns.